### PR TITLE
Docs: Security in Langfuse

### PR DESCRIPTION
**Pull request by lydiagyou**

### CODE REVIEW

File: pages/docs/security.mdx [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+description: Survey of common security problems facing LLM-based applications and how to use Langfuse to trace, prevent, and evaluate security risks.
+---
+
+# Security with Langfuse
+
+There are a host of potential security risks involved with LLM-based applications, such as prompt injection, personally identifiable information (PII), or harmful prompts. Langfuse provides a comprehensive and easy way to monitor and protect against these security risks.
```
Comment (clemra) [inline pages/docs/security.mdx]: 'leakage of' PII

Comment (clemra) [reply pages/docs/security.mdx]: Langfuse provides a comprehensive and easy way to monitor and protect against these security risks.

to 

Langfuse can be used to monitor and protect against these security risks.

File: pages/docs/security.mdx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+This document will provide an overview of how Langfuse can be used with security tools. In the example below we use a [LLM Guard](https://llm-guard.com/), but there are many open-source and/or paid security tools available, such as Prompt Armor, Microsoft Azure Responsible AI, and Lakera Guard.
```
Comment (clemra) [inline pages/docs/security.mdx]: In the example below we use the open source library [LLM Guard](https://llm-guard.com/), but there are other open-source and/or paid security tools available, such as Prompt Armor, Nemo Guardrails, Microsoft Azure Responsible AI, and Lakera Guard

File: pages/docs/security.mdx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  See the trace generated from this example. You can track each step--Anonymize, Summarize, and Deanonymize-- separately: https://cloud.langfuse.com/project/clvh71wlt0000wpuqhferayy5/traces/d31a6a8a-03b4-4487-8d87-40c3a68464a5?observation=aa28105d-c33e-4333-828a-d916de48e387
+</Callout>
+
+
+
+## Advanced Usage
+
+### Langfuse Model-Based Evaluations
+
+You can use a security tool like LLM Guard to scan model prompts and completions, and then use Langfuse to track and evaluate this data after the call is executed.
```
Comment (clemra) [inline pages/docs/security.mdx]: To provide another layer of monitoring after the fact, you can first use a tool like xxxx...... You can additionally add Langfuse to evaluate the responses that were eventually submitted to your users to be made aware of potential risks and identify any gaps in your LLM security setup

File: pages/docs/security.mdx [hunk 4]
```diff
@@ -0,0 +1,114 @@
+---
+description: Survey of common security problems facing LLM-based applications and how to use Langfuse to trace, prevent, and evaluate security risks.
```
Comment (clemra) [inline pages/docs/security.mdx:2]: Nice, let's also prominently link out to the cookbook here.

File: pages/docs/security.mdx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+# Security with Langfuse
+
+There are a host of potential security risks involved with LLM-based applications, such as prompt injection, personally identifiable information (PII), or harmful prompts. Langfuse provides a comprehensive and easy way to monitor and protect against these security risks.
+
+This document will provide an overview of how Langfuse can be used with security tools. In the example below we use a [LLM Guard](https://llm-guard.com/), but there are many open-source and/or paid security tools available, such as Prompt Armor, Microsoft Azure Responsible AI, and Lakera Guard.
+
+## Overview
+
+Short video here?
```
Comment (clemra) [inline pages/docs/security.mdx]: 👍

File: pages/docs/security.mdx [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+- Full name
+- Phone number
+- Email address
+- Social Security number
+- IP Address
+
+Exposing PII to LLMs can pose serious security and privacy risks, such as violating contractual obligations or regulatory compliance requirements, or mitigating the risks of data leakage or a data breach.
+
+The example below shows a simple application that summarizes a given court transcript. For privacy reasons, the application wants to anonymize PII before the information is fed into the model, and then un-redact the response to produce a coherent summary.
+
```
Comment (clemra) [inline pages/docs/security.mdx:51]: Add what other use cases there are for Security within LLM Guard and link out to where to find them

File: pages/docs/security.mdx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+## Advanced Usage
+
+### Langfuse Model-Based Evaluations
+
+You can use a security tool like LLM Guard to scan model prompts and completions, and then use Langfuse to track and evaluate this data after the call is executed.
+
+One way is to understand which flags (toxicity, bias, relevance) are set off the most, which can help teams identify which security risks are most pressing and build more robust tools against those specific issues.
+
+Another way is to monitor the latency and accuracy of different security tools. Langfuse allows users to see how much time each step the security process takes. Another way Langfuse can be used is allowing users to monitor the risk scores returned by certain scanners and using that information to calibrate the sensitivity of their models and fine-tune their models to protect against known security risks.
```
Comment (clemra) [inline pages/docs/security.mdx]: Good one! I'd frame this as an add on under a subheading. sth. like: track latency impact of adding a security library to your LLM app

File: pages/docs/security.mdx [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+
+
+## Advanced Usage
+
+### Langfuse Model-Based Evaluations
+
+You can use a security tool like LLM Guard to scan model prompts and completions, and then use Langfuse to track and evaluate this data after the call is executed.
+
+One way is to understand which flags (toxicity, bias, relevance) are set off the most, which can help teams identify which security risks are most pressing and build more robust tools against those specific issues.
```
Comment (clemra) [inline pages/docs/security.mdx]: link out to langfuse model based evals

Comment (clemra) [inline pages/docs/security.mdx]: Basically the model based eval scores give you a pointer where to look more closely and where you may need to fine tune your security set-up. if you have a lot of traffic it's impractical to look at everything yourself so you need to narrow the sample

File: pages/docs/security.mdx (hunk 4 above)
Comment (lydiagyou) [reply pages/docs/security.mdx:2]: Will do once the cookbook is finalized and added to the repo!

File: pages/docs/security.mdx [hunk 9]
```diff
@@ -0,0 +1,121 @@
+---
+description: Survey of common security problems facing LLM-based applications and how to use Langfuse to trace, prevent, and evaluate security risks.
+---
+
+# Security with Langfuse
+
+There are a host of potential security risks involved with LLM-based applications, such as prompt injection, leakage of personally identifiable information (PII), or harmful prompts. Langfuse can be used to monitor and protect against these security risks.
+
```
Comment (marcklingen) [inline pages/docs/security.mdx:15]: I'd add a second paragraph here to outline how Langfuse + security packages are used in tandem

langfuse
- track security mechanisms to debug and observe whether they work as expected
- track security metrics
- fully async, does not interrupt program execution

**Pull request merged**